doc/cpp_simulation.rst

```diff
@@ -0,0 +1,139 @@
+Simulations
+===========
+
+From recipe to simulation
+-------------------------
+
+To build a simulation the following are needed:
+
+    * An :cpp:class:`arb::recipe` that describes the cells and connections
+      in the model.
+    * An :cpp:class:`arb::hw::node_info` type that describes the hardware
+      resources on which the model will be run.
+
+The workflow to build a simulation is to first generate a
+:cpp:class:`arb::domain_decomposition` that describes the distribution of the model
+over the hardware, then build the simulation.
+
+.. container:: example-code
+
+    .. code-block:: cpp
+
+        // Make description of the hardware that the simulation will run on.
+        arb::hw::node_info node;
+        node.num_cpu_cores = arb::threading::num_threads();
+        node.num_gpus = arb::hw::num_gpus()>0? 1: 0; // use 1 GPU if any available
+
+        // Make a recipe of user defined type my_recipe.
+        my_recipe recipe;
+
+        // Get a description of the partition the model over the cores
+        // (and gpu if available) on node.
+        arb::domain_decomposition decomp = arb::partition_load_balance(recipe, node);
+
+        // Instatitate the simulation.
+        arb::simulation sim(recipe, decomp);
+
+
+Class Documentation
+-------------------
+
+.. cpp:namespace:: arb
+
+.. cpp:class:: simulation
+
+    The executable form of a model. A simulation is constructed
+    from a recipe, and then used to update and monitor model state.
+
+    Simulations take the following inputs:
+
+        * The **constructor** takes an :cpp:class:`arb::recipe` that describes
+          the model, and an :cpp:class:`arb::domain_decomposition` that
+          describes how the cells in the model are assigned to hardware resources.
+        * **Experimental inputs** that can change between model runs, such
+          as external spike trains.
+
+    Simulations provide an interface for executing and interacting with the model:
+
+        * **Advance model state** from one time to another and reset model
+          state to its original state before simulation was started.
+        * **I/O** interface for sampling simulation variables (e.g. compartment voltage
+          and current) and spike output.
+
```

Comment:
'Simulation variable' is a new concept here which is only vaguely defined. How about an 'interface for monitoring simulation state during execution'?